contracts/staking/stakeManager/StakeManager.sol

```diff
@@ -485,7 +492,10 @@ contract StakeManager is IStakeManager, StakeManagerStorage {
         onlyStaker(validatorId)
     {
         address _signer = pubToAddress(signerPubkey);
-        require(_signer != address(0x0) && signerToValidator[_signer] == 0);
+        require(_signer != address(0x0) && signerToValidator[_signer] == 0, "Invalid Signer!");
+        if (latestSignerUpdateEpoch[validatorId] > 0) {
```

Comment:
why this check? it must've been already initialized in `stake/stakeFor` right? please remove this.